### ISSUE TITLE
Could not start an app after installing a NuGet package

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#2668]
After installing Newtonsoft.Json from NuGet package application fails to start with the following exception.

```
Error	1	A locally deployed strong-named assembly was detected. Assembly: Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed. Consider excluding this file by adding a "weaver.ignore" file to your project. (ScErrWeaverFailedStrongNameAsm (SCERR2143))

(Double-click here for additional help)		0	0	

```

The app starts if I manually add `weaver.ignore` file:
```
Newtonsoft.Json.dll
```

https://github.com/Starcounter/Starcounter/wiki/Weaver

Could we automatically add/modify this file when a NuGet package installed?

**Comment by per-samuelsson** [comment]
I have considered this issue some, and I can't come to any really good fits-for-all solution.

> Could we automatically add/modify this file when a NuGet package installed?

Technically, sure. But I'm not sure it's the right thing to do. For once, why should we treat binaries coming from NuGet any different than other binaries? And also, at least ahead in time, there might be binaries  that actually DO contain Starcounter code / database classes that come packaged up in NuGet packs. Disabling them would be a bad thing to do.

One idea I've had is that instead of making a locally depoyed **strong-named assembly** an error (like in your original post), we change it to a warning. And exclude it from being weaved. That way, things would work a little better out of the box, even if it might result in some other confusion (if the assembly actually was designed as a Starcounter library).

**Comment by miyconst** [comment]
Okay, then at least we could add `weaver.ignore` file to project template with some comments inside.

**Comment by per-samuelsson** [comment]
> Okay, then at least we could add weaver.ignore file to project template with some comments inside.

Our aim historically have been to always try to include as little as possible in the standard templates, in order to have such templates keep focus on the very most basic things. Especially to keep it down on things being configurable, since they tend to distract new users.

Maybe we need to have some kind of meeting regarding different templates and what should be included in which (also considering related #2667).

**Comment by miyconst** [comment]
I think the issue can be closed. If someday we come up with some solution, then we can open it again. For now it is okay to manually add the `weaver.ignore` file.
